app/uk/gov/hmrc/individualsemploymentsapi/service/v2/CacheService.scala

```diff
@@ -56,11 +58,20 @@ trait CacheIdBase {
   val id: String
 
   override def toString: String = id
+
+  def encodeVal(toEncode: String): String =
+    BaseEncoding.base64().encode(toEncode.getBytes(Charsets.UTF_8))
 }
 
-case class CacheId(matchId: UUID, interval: Interval, fields: String) extends CacheIdBase {
+case class CacheId(matchId: UUID, interval: Interval, fields: String, empRef: Option[String] = None) extends CacheIdBase {
 
-  lazy val id: String =
-    s"$matchId-${interval.getStart}-${interval.getEnd}-$fields"
+  lazy val id: String = empRef match {
+    case Some(value) => {
+      val encoded = encodeVal(value)
+      s"$matchId-${interval.getStart}-${interval.getEnd}-$fields-$encoded"
```

Comment:
I would rather have the encoded logic outside and have a condition mapping add as that way the "string build up" is in a single place and if it's format changes, it's in 1 location. Nothing major but would be a personal preference. 

E.g.

  val empRefKey = empRef match {
    case Some(value) => s"-${encodeVal(value)}"
    case None => ""
  }
  
  lazy val id: String = s"$matchId-${interval.getStart}-${interval.getEnd}-$fields$empRefKey"

Reply:
If it fails tests just keep this in there. Just a preference thing.

Reply:
yeah, I get you - I like that

Reply:
will sort it now